packages/react/src/button/Button.tsx

```diff
@@ -48,11 +55,60 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   ) => {
     const Comp = asChild ? Slot : "button";
 
-    const presetProps = presets[preset];
+    const presetProps = appearances[appearance];
     const finalColorScheme = colorScheme ?? presetProps.colorScheme;
     const finalVariant = variant ?? presetProps.variant;
-
     const isDisabled = disabled || isLoading;
+    let isIcon = false;
+
+    let content;
+    if (children) {
+      children =
+        asChild && isValidElement(children) ? (
+          cloneElement(
+            children,
+            undefined,
+            <Text
+              as="span"
+              fontSize="inherit"
+              style={{
+                paddingInline: "4px",
+              }}
+            >
+              {children.props.children}
+            </Text>,
+          )
+        ) : (
+          <Text
+            as="span"
+            fontSize="inherit"
+            style={{
+              paddingInline: "4px",
+            }}
+          >
+            {children}
+          </Text>
+        );
```

Comment:
is it possible to avoid this `children` wrapping? this won't work in all cases and its best if we can just avoid it completely

Reply:
Should I completely remove this bhai?
It is affecting the `Link` story. Other than that, everything else looks the same

Reply:
removed the `children` wrapping.